Config/SEGGER_RTT_Conf.h

```diff
@@ -147,15 +147,17 @@ Revision: $Rev: 21386 $
 */
 #if ((defined(__SES_ARM) || defined(__SES_RISCV) || defined(__CROSSWORKS_ARM) || defined(__GNUC__) || defined(__clang__)) && !defined (__CC_ARM) && !defined(WIN32))
   #if defined(__ZEPHYR__) && defined (CONFIG_SEGGER_RTT_CUSTOM_LOCKING)
-    #include <zephyr/kernel.h>
     #ifdef CONFIG_MULTITHREADING

```

Comment:
This file does not include any Zephyr headers, yet it is referencing the Kconfig definitions such as `CONFIG_MULTITHREADING`.

How is this supposed to work? Should it at least include `autoconf.h`?

Reply:
I removed the autoconf.h include according to [this comment](https://github.com/zephyrproject-rtos/zephyr/pull/48513#discussion_r934510145). Apperently the build magic does the job.

Reply:
That is specific to the Zephyr build process and does not necessarily apply to the module build process.

But, for this particular module, it looks like it is being built inside the Zephyr build system, so I suppose this is ok.